src/providers/authentication/authentication.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import { MSAdal, AuthenticationContext, AuthenticationResult } from '@ionic-native/ms-adal';
+import { InAppBrowser, InAppBrowserOptions } from '@ionic-native/in-app-browser';
+import { AppConfigProvider } from '../app-config/app-config';
+
+@Injectable()
+export class AuthenticationProvider {
+
+  public authenticationSettings: any;
+  
+  private authenticationToken: string;
+
+  constructor(
+    private msAdal: MSAdal,
+    private inAppBrowser: InAppBrowser,
+    appConfig: AppConfigProvider) {
+    this.authenticationSettings = appConfig.getAppConfig().authentication;
+  }
+
+  isAuthenticated = (): boolean => {
+    return this.authenticationToken ? true : false;
```

Comment:
Need to have a think about how to make this a better check for whether a user is logged in or not, should have a look in the adal library's source code to see if there's anything we can use there. Failing that, we could decode the token and check for some essential values.

Reply:
Agreed - This is something we need to investigate - could it be worth spinning this off into a second story - Azure AD Token Validation in Mobile App . I'm wary of pushing more into this  ticket